scilpy/image/operations.py

```diff
@@ -0,0 +1,461 @@
+
+
+def subtraction(input_list):
+    """
+    subtraction: IMG_1 IMG_2
+        Subtract two images together (IMG_1 - IMG_2).
+    """
+    _validate_length(input_list, 2)
+    _validate_arrays(*input_list)
+
+    return input_list[0] - input_list[1]
+
+
+def multiplication(input_list):
+    """
+    multiplication: IMGs
+        Multiply multiple images together (danger of underflow and overflow)
+    """
+    _validate_length(input_list, 2, atleast=True)
+    _validate_arrays(*input_list)
+
+    output_data = input_list[0]
+    for data in input_list[1:]:
+        output_data *= data
+
+    return output_data
+
+
+def division(input_list):
+    """
+    division: IMG_1 IMG_2
+        Divide two images together (danger of underflow and overflow)
+    """
+    _validate_length(input_list, 2)
+    _validate_arrays(*input_list)
+    ref_array = input_list[0]
+
+    output_data = np.zeros(ref_array.shape)
+    output_data[input_list[1] > 0] = input_list[0][input_list[1] > 0] \
+        / input_list[1][input_list[1] > 0]
+    return output_data
+
+
+def mean(input_list):
+    """
+    mean: IMGs
+        If a single 4D image is provided, average along the last dimension.
+    """
+    _validate_length(input_list, 1, atleast=True)
+    _validate_arrays(*input_list)
+    ref_array = input_list[0]
+
+    if len(input_list) == 1 and not ref_array.ndim > 3:
+        logging.error('This operation with only one operand requires 4D data.')
+        raise ValueError
+
+    in_data = np.squeeze(np.rollaxis(np.array(input_list), 0,
+                                     input_list[0].ndim+1))
+
+    return np.average(in_data, axis=-1)
+
+
+def std(input_list):
+    """
+    std: IMGs
+        Compute the standard deviation average of multiple images.
+        If a single 4D image is provided, compute the STD along the last
+        dimension.
+    """
+    _validate_length(input_list, 1, atleast=True)
+    _validate_arrays(*input_list)
+    ref_array = input_list[0]
+
+    if len(input_list) == 1 and not ref_array.ndim > 3:
+        logging.error('This operation with only one operand requires 4D data.')
+        raise ValueError
+
+    in_data = np.squeeze(np.rollaxis(np.array(input_list), 0,
+                                     input_list[0].ndim+1))
+
+    return np.std(in_data, axis=-1)
+
+
+def union(input_list):
+    """
+    union: IMGs
+        Binary operation to keep voxels that are in any file.
+    """
+    output_data = addition(input_list)
+    output_data[output_data != 0] = 1
+
+    return output_data
+
+
+def intersection(input_list):
+    """
+    intersection: IMGs
+        Binary operation to keep the voxels that are present in all files.
+    """
+    output_data = multiplication(input_list)
+    output_data[output_data != 0] = 1
+
+    return output_data
+
+
+def difference(input_list):
+    """
+    difference: IMG_1 IMG_2
+        Binary operation to keep voxels from the first file that are not in
+        the second file.
+    """
```

Comment:
Can't find a better way to say it but I feel we can't say that a voxel "doesn't exist". It's non zero.